cli/lib/site-utils.ts

```diff
@@ -83,20 +83,31 @@ export async function setupCustomDomain(
  * @param stoppedSiteId - The ID of the site that was just stopped (to exclude from the check)
  */
 export async function stopProxyIfNoSitesNeedIt(
-	stoppedSiteId: string,
+	stoppedSiteIds: string | string[],
 	logger: Logger< LoggerAction >
 ): Promise< void > {
+	const stoppedSiteIdsArray = Array.isArray( stoppedSiteIds ) ? stoppedSiteIds : [ stoppedSiteIds ];
+
 	const proxyProcess = await isProxyProcessRunning();
 	if ( ! proxyProcess ) {
 		return;
 	}
 
 	const appdata = await readAppdata();
 
-	for ( const site of appdata.sites ) {
-		if ( site.id !== stoppedSiteId && site.customDomain && ( await isServerRunning( site.id ) ) ) {
-			return;
-		}
+	const remainingSites = appdata.sites.filter(
+		( site ) => ! stoppedSiteIdsArray.includes( site.id )
+	);
+
+	// Filter to sites with custom domains that might need the proxy
+	const sitesWithCustomDomains = remainingSites.filter( ( site ) => site.customDomain );
+
+	const sitesStillRunning = await Promise.all(
+		sitesWithCustomDomains.map( ( site ) => isServerRunning( site.id ) )
+	);
```

Comment:
```suggestion
	const remainingSitesWithCustomDomains = appdata.sites.filter(
		( site ) => ! stoppedSiteIdsArray.includes( site.id ) && site.customDomain
	);

	const sitesStillRunning = await Promise.all(
		remainingSitesWithCustomDomains.map( ( site ) => isServerRunning( site.id ) )
	);
```
I actually think this is clearer if the variable name is more verbose and we collapse the two first `filter` calls into a single one, but feel free to make your own call here.